### PR TITLE
Implement QDir in rbx_xml

### PR DESCRIPTION
**Pull request by kennethloeffler**
`generate_reflection` needs to be able to access Roblox Studio's auto-save path (see #181). It is stored in what is essentially a `.rbxmx` file in Roblox's installation root. Here's a piece of trivia for you: Roblox Studio will correctly read a path delimited with any number of (possibly interleaved) forward and back slashes, but always writes it back delimited with one forward slash.

### CODE REVIEW

File: rbx_types/src/variant.rs [hunk 1]
```diff
@@ -126,6 +126,7 @@ make_variant! {
     Vector3(Vector3),
     Vector3int16(Vector3int16),
     OptionalCFrame(Option<CFrame>),
+    QDir(std::path::PathBuf),
```
Comment (LPGhatguy) [inline rbx_types/src/variant.rs:129]: Should we newtype this? I'm a little afraid of the format eventually growing a not-weird-Qt-type path. We can sort of follow the lead from `Content` here.

**Review by LPGhatguy — CHANGES_REQUESTED** [review]
Easy implementation, question about future compatibility.

Can you also update the relevant CHANGELOG files?

**Comment by kennethloeffler** [comment]
I just scattershot implemented various traits - let me know if you think we need more!

File: rbx_types/src/qdir.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+impl FromStr for QDir {
+    type Err = Infallible;
+
+    fn from_str(path: &str) -> Result<Self, Infallible> {
+        Ok(Self {
+            path_buf: PathBuf::from(path),
+        })
+    }
+}
```
Comment (LPGhatguy) [inline rbx_types/src/qdir.rs:106]: Let's narrow these down to just `From<&Path>` and `From<PathBuf>`. Every impl we have here is another thing we have to uphold for forward compatibility and I think keeping a super niche type like this slim is probably okay.

File: rbx_types/src/qdir.rs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+impl Deref for QDir {
+    type Target = Path;
+
+    fn deref(&self) -> &Path {
+        self.path_buf.as_path()
+    }
+}
```
Comment (LPGhatguy) [inline rbx_types/src/qdir.rs:56]: I don't think a deref here is all that important. We can `AsRef` into `Path`, we're not making a total drop-in replacement.

File: rbx_types/src/qdir.rs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    pub fn into_path_buf(self) -> PathBuf {
+        self.path_buf
+    }
+}
+
+impl AsRef<OsStr> for QDir {
+    fn as_ref(&self) -> &OsStr {
+        self.path_buf.as_os_str()
+    }
+}
```
Comment (LPGhatguy) [inline rbx_types/src/qdir.rs:36]: I don't think we need this impl

File: rbx_xml/src/types/qdir.rs [hunk 5]
```diff
@@ -0,0 +1,42 @@
+use std::io::{Read, Write};
+
+use crate::{
+    core::XmlType,
+    deserializer_core::XmlEventReader,
+    error::{DecodeError, EncodeError},
+    serializer_core::XmlEventWriter,
+};
+
+use rbx_dom_weak::types::QDir;
```
Comment (LPGhatguy) [inline rbx_xml/src/types/qdir.rs:10]:
```suggestion
use rbx_dom_weak::types::QDir;

use crate::{
    core::XmlType,
    deserializer_core::XmlEventReader,
    error::{DecodeError, EncodeError},
    serializer_core::XmlEventWriter,
};
```

We usually order external crates above internal imports.

**Review by LPGhatguy — CHANGES_REQUESTED** [review]
Too many impls! I think we can cut down on these substantially because of how niche of a type this is.

**Comment by kennethloeffler** [comment]
Further work on #181 shows this to be unnecessary.

**Pull request closed without merging**